setup/src/Magento/Setup/Console/Command/DiCompileCommand.php

```diff
@@ -156,20 +165,33 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
 
         $modulePaths = $this->componentRegistrar->getPaths(ComponentRegistrar::MODULE);
+        $moduleStatuses = $this->deploymentConfig->get(ConfigOptionsListConstants::KEY_MODULES);
+        if (!$moduleStatuses || !is_array($moduleStatuses)) {
+            return Cli::RETURN_FAILURE;
+        }
+        $enabledModuleStatuses = array_filter($moduleStatuses, function ($enabled) {
+            return $enabled === 1;
+        });
+        $enabledModules = array_keys($enabledModuleStatuses);
+
+        $modulePathsEnabled = array_filter($modulePaths, function ($path, $module) use ($enabledModules) {
+            return in_array($module, $enabledModules, true);
+        }, ARRAY_FILTER_USE_BOTH);
```

Comment:
This can be simplified like this:
```suggestion
        $modulePathsEnabled = array_filter($modulePaths, function ($path, $module) use ($moduleStatuses) {
            return ($moduleStatuses[$module] ?? 0) === 1;
        }, ARRAY_FILTER_USE_BOTH);
```

Reply:
Done, thank you